wfe2/wfe.go

```diff
@@ -1631,6 +1647,28 @@ func (wfe *WebFrontEndImpl) Certificate(ctx context.Context, logEvent *web.Reque
 			), nil)
 			return
 		}
+
+		// If the requested chain is outside the bounds of the available chains,
+		// then it is an error by the client - not found.
+		if requestedChain < 0 || requestedChain >= len(availableChains) {
+			wfe.sendError(response, logEvent, probs.NotFound("Certificate not found"), nil)
```

Comment:
`Unknown issuance chain` might make a bit more sense than `Certificate not found`.